library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -163,12 +175,24 @@ impl Command {
     unsafe fn do_fork(&mut self) -> Result<(pid_t, pid_t), io::Error> {
         use crate::sys::os::errno;
 
-        let mut tries_left = MAX_FORKSPAWN_TRIES;
+        let mut minimum_delay = None;
+        let mut delay = MIN_FORKSPAWN_SLEEP;
+
         loop {
             let r = libc::fork();
-            if r == -1 as libc::pid_t && tries_left > 0 && errno() as libc::c_int == libc::EBADF {
-                thread::yield_now();
-                tries_left -= 1;
+            if r == -1 as libc::pid_t && errno() as libc::c_int == libc::EBADF {
+                if minimum_delay.is_none() {
+                    minimum_delay = Some(get_clock_resolution());
+                }
+                if delay < minimum_delay.unwrap() {
+                    // We cannot sleep this short (it would be longer).
+                    // Yield instead.
+                    thread::yield_now();
+                } else {
+                    thread::sleep(delay);
+                }
+                delay *= 2;
+                continue;
```

Comment:
I would like it if we still bailed for `ErrorKind::WouldBlock` or `ErrorKind::Interrupted` after a while, probably if we're getting up to actual seconds between yields.